src/test/ui/associated-type-bounds/nested-lifetime-bounds.rs

```diff
@@ -0,0 +1,23 @@
+// compile-fail
+
+use std::fmt::Debug;
+
+trait Lam<Binder> { type App; }
+
+fn nested_bounds<_0, _1, _2, D>()
+where
+    D: Clone + Iterator<Item: Send + for<'a> Iterator<Item: for<'b> Lam<&'a &'b u8, App = _0>>>,
```

Comment:
Yeah I know. I need help on how to detect this. The problem is that the checking done right now in `librustc/middle` (a mysterious module to me to begin with) looks at the HIR, and the desugaring for this is done during typeck.

Reply:
This and the non-ideal message for E0277 (`T = impl Bounds` rather than `T: Bounds`) should be solved in a follow-up PR. Unfortunately span information is not available at the time this error is emitted, but @estebank should have some ideas hopefully.

Reply:
@aturon This is the one outstanding issue.